packages/server/src/course/course.service.ts

```diff
@@ -234,4 +236,63 @@ export class CourseService {
       );
     }
   }
+
+  async getUserInfo(
+    courseId: number,
+    page: number,
+    pageSize: number,
+    role?: Role,
+    search?: string,
+  ): Promise<UserPartial[]> {
+    try {
+      const query = await getRepository(UserModel)
+        .createQueryBuilder()
+        .leftJoin(
+          UserCourseModel,
+          'UserCourseModel',
+          '"UserModel".id = "UserCourseModel"."userId"',
+        )
+        .where('"UserCourseModel"."courseId" = :courseId', { courseId });
+
+      // check if searching for specific role
+      if (role) {
+        query.andWhere('"UserCourseModel".role = :role', { role });
+      }
+      // check if searching for specific name
+      if (search) {
+        const likeSearch = `%${search}%`.toUpperCase();
+        query.andWhere(
+          new Brackets((q) => {
+            q.where('UPPER("UserModel"."firstName") like :firstSearch', {
+              firstSearch: likeSearch,
+            }).orWhere('UPPER("UserModel"."lastName") like :lastSearch', {
+              lastSearch: likeSearch,
+            });
+          }),
+        );
+      }
+
+      // run query
+      const users = query
+        .select([
+          'UserModel.id',
+          'UserModel.firstName',
+          'UserModel.lastName',
+          'UserModel.photoURL',
+          'UserModel.email',
+        ])
+        .orderBy('UserModel.firstName')
+        .skip((page - 1) * pageSize)
+        .take(pageSize)
+        .getMany();
+
+      return users;
+    } catch (err) {
+      console.error(err);
+      throw new HttpException(
+        ERROR_MESSAGES.common.pageOutOfBounds,
```

Comment:
hmm, will the error always be a `pageOutOfBounds`?

Reply:
no, @sumitde22 brings up a good point below too, I don't think it can ever be an out of bounds page anyways because the way the query is written it won't error out it will just return no students.